language/move-prover/stackless-bytecode-generator/src/stackless_bytecode_generator.rs

```diff
@@ -792,4 +798,274 @@ impl<'a> StacklessBytecodeGenerator<'a> {
             _ => sig.clone(),
         }
     }
+
+    /// Remove MoveLoc, StLoc and CopyLoc from Stackless Bytecode
+    pub fn simplify_bytecode(code: &[StacklessBytecode]) -> Vec<StacklessBytecode> {
+        let mut new_code = vec![];
+        let mut new_offsets = BTreeMap::new(); // Used later to decide the new offset to branch to
+        let mut equiv_temps = BTreeMap::new(); // Stores temp->local mappings
+        let mut new_offset = 0;
+
+        for (i, bytecode) in code.iter().enumerate() {
```

Comment:
This seems to assume we have single-assignment in bytecode. Do we actually have? I recall having something like `let x = if (c) e1 else e2; ...` in Move and it wasn't SA, but I didn't follow up on deeper analysis. In any case, we can settle this after this PR once we activate this.

Reply:
Depending on how you define single-assignment, but I think we do have it in stackless bytecode since the added temps are all disposable.

Reply:
And because of this I didn't feel the need to do any reaching definitions analysis.

Reply:
Temporaries might be assigned once only, but in different execution paths. Example:

```
       BranchIf(l2, c)
       MoveLoc(t, x)
       Branch(l3)
l2:   MoveLoc(t, y)
l3:   ...
```

As far as I understand, your algorithm would always alias `t` with `y`, independent of the value of `c`.

In the reaching definition scenario, no definition of `t` would reach `l3`, so you would not be able to eliminate this temporary. Please correct me if I'm wrong.

In general, it might be saver here to stick to standard techniques, even though they might be more general as we need.